### ISSUE TITLE
"Could not find com.chaquo.python.runtime:chaquopy_java" in Flutter plugin project

### ISSUE THREAD

**Issue by mhsmith** [chaquo/chaquopy#615]
_Originally posted by @arnavpon in https://github.com/chaquo/chaquopy/issues/94#issuecomment-1114240902_

Hi Malcolm, 

I'm building a Flutter plugin that requires chaquopy as a dependency, but having a lot of trouble correctly configuring the build.gradle file. The main issue is that there's only one build.gradle in a Flutter plugin's android module (there's no app-specific build.gradle), so I tried to add all the necessary configuration to this file. When I try to build, here's the error I'm getting:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':pydroid:bundleDebugAar'.
> Could not resolve all files for configuration ':pythonRuntimeJava'.
   > Could not find com.chaquo.python.runtime:chaquopy_java:11.0.0.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/com/chaquo/python/runtime/chaquopy_java/11.0.0/chaquopy_java-11.0.0.pom
       - https://repo.maven.apache.org/maven2/com/chaquo/python/runtime/chaquopy_java/11.0.0/chaquopy_java-11.0.0.pom
     Required by:
         project :
```

And the build.gradle file:
```
buildscript {
    repositories {
        google()
        mavenCentral()
        maven { url "https://chaquo.com/maven" }
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:4.1.0'
        classpath "com.chaquo.python:gradle:11.0.0"
    }
}

rootProject.allprojects {
    repositories {
        google()
        mavenCentral()
    }
}

apply plugin: 'com.android.library'
apply plugin: 'com.chaquo.python'

android {
    compileSdkVersion 30

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        minSdkVersion 16
        ndk {
            abiFilters "armeabi-v7a", "arm64-v8a", "x86", "x86_64"
        }
        python {
            buildPython "~/opt/anaconda3/envs/chaquopy/bin/python3.8"
            pip {
                install 'numpy'
            }
        }
    }
}
```

Apologies in advance if I'm making an obvious error - I'm new to Android development. Thank you!

**Comment by mhsmith** [comment]
I assume the project is https://github.com/arnavpon/pydroid? If so, can you update it with the current version of your code? You can put it on a separate branch if you prefer.

**Comment by arnavpon** [comment]
Hi Malcolm,

Thank you for getting back to me so quickly. I just updated the github repo
with the existing code.

**Comment by mhsmith** [comment]
It looks like Chaquopy is confused by the way the Flutter template includes one project into another. When I built the plugin project directly, by running `flutter build aar` in the root directory, everything works fine, though I needed to add the `androidx.annotation` dependency to the build.gradle file. But when I tried to build the example/android project, which includes the plugin project, I got the error message above.

We'll fix this in the next version of Chaquopy. Meanwhile, it looks like you can work around it by editing example/android/build.gradle so that its `buildscript.repositories` block includes the chaquo.com Maven repository, just as you've already done for the plugin project.

**Comment by arnavpon** [comment]
Malcolm,

Thank you for being so responsive. I truly appreciate the help.

Using the workaround you mentioned above, the example app builds
successfully. However, whenever I attempt to call any chaquopy code within
the android plugin, the application crashes with the following exception:

[image: Screen Shot 2022-05-03 at 10.03.57 PM.png]

It's specifically the chaquopy code generating the error, as other data
passed through the method channel, such as identifying the android OS build
version, works as expected. The up-to-date code generating this error is on
the github repository.

**Comment by arnavpon** [comment]
Will do, thanks!

**Comment by mhsmith** [comment]
_Originally posted by @arnavpon in https://github.com/chaquo/chaquopy/issues/94#issuecomment-1118083130_

Malcolm,

Thank you for being so responsive. I'm building a Flutter plugin that utilizes chaquopy. Configuring the module's _build.gradle_ as well as the top-level and app-level _build.gradle_ files in the "example" app as specified in the chaquopy docs enables a successful build. However, whenever I attempt to call any chaquopy code within the android plugin, the application crashes with the following exception:

<img width="747" alt="Screen Shot 2022-05-03 at 10 03 57 PM" src="https://user-images.githubusercontent.com/16437565/166851588-4f5acbd0-6e64-4539-a5c8-0e5615eded2b.png">

It's specifically the chaquopy code generating the error, as other data passed through the method channel, such as identifying the android OS build version, works as expected.

**Comment by mhsmith** [comment]
If you search for this error on Google, it returns https://github.com/flutter/flutter/issues/11104, which suggests that the fix is to replace `ex.getStackTrace()` at PydroidPlugin.java:100 with `null`. So maybe this is not the correct way of passing an exception from Java to Dart. I don't know much about Flutter, so you'd have to search elsewhere for help on that.

**Comment by mhsmith** [comment]
The original issue ("Could not find com.chaquo.python.runtime:chaquopy_java") is fixed in Chaquopy 12.0.0. To upgrade, change the Chaquopy version number in your app's top-level build.gradle file.
